---
title: "Coverage Analysis for MVL eye gene panel for different NGS tech"
output: html_notebook
---

## Load packages
```{r, message = F}
library(tidyverse)
library(data.table)
source('~/git/scripts/theme_Publication.R')
```
## Load in data from coverage_calc.sh
```{r}
depth_filenames <- list.files('../data/processed/', pattern='depth.gz')
depth_filenames <- sapply(depth_filenames, function(x) gsub(".depth.gz","",x))
depth_filepaths <- list.files('../data/processed/', pattern='depth.gz', full.names = T)
depth_filenames
depth_filepaths
dat = list()
for (i in seq(1,length(depth_filenames))){
  temp <- fread(paste('gzcat',depth_filepaths[i])) %>% data.frame()
  colnames(temp) <- c('chr','start','stop','Transcript','Gene','Strand', 'Position','Depth')
  dat[[depth_filenames[i]]] <- temp
}
dat_df <- bind_rows(dat, .id='NGS_file')
rm(dat)
rm(temp)
```


## Calculate overall coverage, split by NGS test

### Coverage Density for Exome/WGS
#### Red line is <=10 reads
```{r}
dat_df %>% filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% group_by(NGS_file) %>% 
  ggplot(aes(x=Depth, colour=NGS_file)) + geom_density() + theme_Publication() + coord_cartesian(xlim=c(0,300)) + geom_vline(xintercept = 10,colour='red')
```

### Coverage Density for Casey/MVL Panels
#### Red line is <=10 reads
```{r}
dat_df %>% filter(!NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% group_by(NGS_file) %>% 
  ggplot(aes(x=log2(Depth+1), colour=NGS_file)) + geom_density() + theme_Publication() + geom_vline(xintercept = log2(10),colour='red')
```
### Coverage Density for Casey/MVL RD with the Exome/WGS
#### Red line is <=10 reads
```{r}
dat_df %>% filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome','16-00307_RD')) %>% group_by(NGS_file) %>% 
  ggplot(aes(x=log2(Depth+1), colour=NGS_file)) + geom_density() + theme_Publication() + geom_vline(xintercept = log2(10), colour='red')
```

## Mean and median coverage for each test
```{r}
dat_df %>% group_by(NGS_file) %>% summarise(Mean=mean(Depth), Median=median(Depth), `25th`=quantile(Depth,0.25), `75th` = quantile(Depth,0.75))
```

## Percent of base pairs with less than N reads
```{r}
dat_df %>% group_by(NGS_file) %>% summarise(`10 reads`= sum(Depth<10)/n() * 100, `15 reads`= sum(Depth<15)/n() * 100, `20 reads`= sum(Depth<20)/n() * 100, `25 reads`= sum(Depth<25)/n() * 100)
```

## Percent of base pairs with less than 10 reads, split by NGS_file and gene
```{r, fig.width=10, fig.height=35}
dat_df %>% group_by(NGS_file, Gene) %>% summarise(`10 reads`= sum(Depth<10)/n() * 100) %>% 
  ggplot(aes(x=Gene, y=`10 reads`)) + geom_bar(stat='identity') + theme_Publication() + coord_flip() +
  facet_wrap(~ NGS_file, ncol=8)
```

## Genes whose coverage improves from exome -> 30X WGS -> 50X WGS
```{r, fig.width=10, fig.height=35}
dat_df %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  arrange(-`Exome -> 30X WGS`)

ordered_genes_exome_30x <- dat_df %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  arrange(`Exome -> 30X WGS`) %>% .[['Gene']]

dat_df %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  arrange(-`30X WGS -> 50X WGS`)

ordered_genes_30x_50x <- dat_df %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  arrange(`30X WGS -> 50X WGS`) %>% .[['Gene']]

# Summary stats:
# Num of genes with a >20% delta improvement going from Exome to 30X:
dat_df %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  filter(`Exome -> 30X WGS` > 20) %>% nrow()
ordered_genes_1 <- 
# Num of genes with a >20% delta improvement going from 30X to 50X:
dat_df %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  filter(`30X WGS -> 50X WGS` > 20) %>% nrow()


```

## Genes whose coverage improves more than 20% from exome -> 30X WGS; plot
```{r, fig.width=5, fig.height=10}
dat_df %>% mutate(Gene = factor(Gene, levels = ordered_genes_exome_30x)) %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  filter(`Exome -> 30X WGS` > 20) %>% 
  gather(Comparison, Value, CCGO_800596_Exome:`30X WGS -> 50X WGS`) %>% 
  filter(Comparison %in% c('Exome -> 30X WGS', 'CCGO_800596_Exome', 'CCGO_800071_WGS')) %>% 
  mutate(Comparison = factor(Comparison, levels = c('Exome -> 30X WGS', 'CCGO_800596_Exome', 'CCGO_800071_WGS'))) %>% 
  ggplot(aes(x=Gene, y=Value))+ geom_bar(stat='identity') + theme_Publication() + coord_flip() +
  facet_wrap(~ Comparison, ncol=3)+ labs(y="% BP under 10 Reads across Gene Body")
```
## Genes who coverage improves more than 20% from 30X WGS -> 50X WGS; plot
```{r, fig.width=5, fig.height=5}
dat_df %>% mutate(Gene = factor(Gene, levels = ordered_genes_30x_50x)) %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  filter(`30X WGS -> 50X WGS` > 20) %>% 
  gather(Comparison, Value, CCGO_800596_Exome:`30X WGS -> 50X WGS`) %>% 
  filter(Comparison %in% c('30X WGS -> 50X WGS', 'CCGO_800071_WGS',  'HG001_WGS')) %>% 
  mutate(Comparison = factor(Comparison, levels = c('30X WGS -> 50X WGS', 'CCGO_800071_WGS',  'HG001_WGS'))) %>% 
  ggplot(aes(x=Gene, y=Value))+ geom_bar(stat='identity') + theme_Publication() + coord_flip() +
  facet_wrap(~ Comparison, ncol=3) + labs(y="% BP under 10 Reads across Gene Body")+ scale_y_continuous(limits = c(0,100))
```
# Exome -> 30x, Developmental Genes
```{r, fig.width=5, fig.height=5}
dev <- c('ABCB6','ALDH1A3','ATOH7','B3GALTL','BCOR','BMP4','c12orf57','CASK','CHD7','COL4A1','COX7B','CRX','CYP1B1','DCDC1','ELP4','FKRP','FKTN','FOXC1','FOXC2','FOXE3','GDF3','GDF6','HESX1','HMGB3','ISPD','LAMB2','LARGE','LHX2','MAB21L2','MFRP','NAA10','NDP','NKX5- 3','OTX2','PAX2','PAX6','PITX2','PITX3','POMT1','POMT2','PRSS56','RAX','RAB3GAP1','RAB3GAP2','RAB18','RARB','SHH','SIX3','SIX6','SLC25A1','SNX3','SOX2','SOX3','STRA6','TENM3','TMEM98','VAX1','VSX2','ZEB2')
dat_df %>% mutate(Gene = factor(Gene, levels = ordered_genes_exome_30x)) %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  filter(Gene %in% dev) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  gather(Comparison, Value, CCGO_800596_Exome:`30X WGS -> 50X WGS`) %>% 
  filter(Comparison %in% c('Exome -> 30X WGS', 'CCGO_800596_Exome', 'CCGO_800071_WGS')) %>% 
  mutate(Comparison = factor(Comparison, levels = c('Exome -> 30X WGS', 'CCGO_800596_Exome', 'CCGO_800071_WGS'))) %>% 
  ggplot(aes(x=Gene, y=Value))+ geom_bar(stat='identity') + theme_Publication() + coord_flip() +
  facet_wrap(~ Comparison, ncol=3)+ labs(y="% BP under 10 Reads across Gene Body")
```

# 30x -> 50x, Developmental Genes
```{r, fig.width=5, fig.height=5}

dat_df %>% mutate(Gene = factor(Gene, levels = ordered_genes_30x_50x)) %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  filter(Gene %in% dev) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  gather(Comparison, Value, CCGO_800596_Exome:`30X WGS -> 50X WGS`) %>% 
  filter(Comparison %in% c('30X WGS -> 50X WGS', 'CCGO_800071_WGS',  'HG001_WGS')) %>% 
  mutate(Comparison = factor(Comparison, levels = c('30X WGS -> 50X WGS', 'CCGO_800071_WGS',  'HG001_WGS'))) %>% 
  ggplot(aes(x=Gene, y=Value))+ geom_bar(stat='identity') + theme_Publication() + coord_flip() +
  facet_wrap(~ Comparison, ncol=3) + labs(y="% BP under 10 Reads across Gene Body") + scale_y_continuous(limits = c(0,100))
```

# Exome -> 30x, Pigmentation Genes
```{r, fig.width=5, fig.height=3}
pig <- c('OA1','OCA1','OCA2','OCA3','OCA4','c10orf11','HPS1','HPS2','HPS3','HPS4','HPS5','HPS6','HPS7','HPS8','HPS9','SLC24A5','SLC38A8','SLC45A2','LYST','EDN3','EDNRB','MITF','PAX3','RET','SNAI2','SOX10','KIT','MLPH','MYO5A','RAB27A')
dat_df %>% mutate(Gene = factor(Gene, levels = ordered_genes_exome_30x)) %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  filter(Gene %in% pig) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  gather(Comparison, Value, CCGO_800596_Exome:`30X WGS -> 50X WGS`) %>% 
  filter(Comparison %in% c('Exome -> 30X WGS', 'CCGO_800596_Exome', 'CCGO_800071_WGS')) %>% 
  mutate(Comparison = factor(Comparison, levels = c('Exome -> 30X WGS', 'CCGO_800596_Exome', 'CCGO_800071_WGS'))) %>% 
  ggplot(aes(x=Gene, y=Value))+ geom_bar(stat='identity') + theme_Publication() + coord_flip() +
  facet_wrap(~ Comparison, ncol=3)+ labs(y="% BP under 10 Reads across Gene Body") + scale_y_continuous(limits = c(0,100))
```

# 30x -> 50x, Pigmentation Genes
```{r, fig.width=5, fig.height=3}

dat_df %>% mutate(Gene = factor(Gene, levels = ordered_genes_30x_50x)) %>% 
  filter(NGS_file %in% c('CCGO_800071_WGS','HG001_WGS','CCGO_800596_Exome')) %>% 
  filter(Gene %in% pig) %>% 
  group_by(NGS_file, Gene) %>% 
  summarise(`% <10 reads`= sum(Depth<10)/n() * 100) %>% 
  spread(NGS_file, `% <10 reads`) %>% 
  select(Gene, CCGO_800596_Exome, CCGO_800071_WGS, HG001_WGS) %>% 
  mutate(`Exome -> 30X WGS` = CCGO_800596_Exome - CCGO_800071_WGS, `30X WGS -> 50X WGS` = CCGO_800071_WGS - HG001_WGS) %>% 
  gather(Comparison, Value, CCGO_800596_Exome:`30X WGS -> 50X WGS`) %>% 
  filter(Comparison %in% c('30X WGS -> 50X WGS', 'CCGO_800071_WGS',  'HG001_WGS')) %>% 
  mutate(Comparison = factor(Comparison, levels = c('30X WGS -> 50X WGS', 'CCGO_800071_WGS',  'HG001_WGS'))) %>% 
  ggplot(aes(x=Gene, y=Value))+ geom_bar(stat='identity') + theme_Publication() + coord_flip() +
  facet_wrap(~ Comparison, ncol=3) + labs(y="% BP under 10 Reads across Gene Body") + scale_y_continuous(limits = c(0,100))
```

## Density in coding exons, split by gene, coloured by test
### (I can add any gene you want here)
```{r}
dat_df %>% filter(Gene %in% c('ABCA4', 'SAG', 'CHM')) %>% group_by(Gene, NGS_file) %>% 
  ggplot(aes(x=log10(Depth+1),colour=NGS_file)) + 
  geom_density() + 
  theme_Publication() + 
  facet_wrap(~ Gene, scales = 'free_y', ncol = 4)
```

## Coverage, for a few genes, across the gene body (areas with no dots are generally introns)
### (I can substitute other genes)
```{r, fig.width=4,fig.height=6}
dat_df %>% filter(Gene %in% c('ABCA4','SAG','CHM')) %>% group_by(NGS_file,Gene) %>% mutate(Position=start+Position) %>% 
  ggplot(aes(y=log2(Depth+1),x=Position,colour=NGS_file)) + geom_jitter(alpha=0.4,size=0.1) + theme_Publication() + facet_wrap(~Gene, scale='free_x', ncol=1) + geom_hline(yintercept=log2(10),colour='red') + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```
